### PR TITLE
fix: use pointer for minor in device crd

### LINKED ISSUE #655: [BUG] minor in device crd is empty when it should be 0
**What happened**:
image

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- App version:
- Kubernetes version (use `kubectl version`):
- Install details (e.g. helm install args):
- Others:

### PR DESCRIPTION
**Pull request by jasonliu747**
Signed-off-by: Jason Liu <jasonliu747@gmail.com>

### Ⅰ. Describe what this PR does
Use a pointer for the minor field, so that the zero value of the JSON type can be differentiated from the missing value.

### Ⅱ. Does this pull request fix one issue?
fixes #655 

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

**Comment by hormes** [comment]
@buptcozy please review this pr

**Comment by jasonliu747** [comment]
/hold
CI keeps failing, havn't figured it out yet.

File: pkg/koordlet/statesinformer/states_device_linux.go [hunk 1]
```diff
@@ -145,7 +145,7 @@ func (s *statesInformer) buildGPUDevice() []schedulingv1alpha1.DeviceInfo {
 		s.gpuMutex.RUnlock()
 		deviceInfos = append(deviceInfos, schedulingv1alpha1.DeviceInfo{
 			UUID:   gpu.DeviceUUID,
-			Minor:  gpu.Minor,
+			Minor:  &gpu.Minor,
```
Comment (saintube) [inline pkg/koordlet/statesinformer/states_device_linux.go:148]: do not use the pointer of a for-range value

Comment (jasonliu747) [reply pkg/koordlet/statesinformer/states_device_linux.go:149]: everyday you learn something new

**Review by buptcozy — APPROVED** [review]
/LGTM

**Pull request merged**